pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -347,27 +347,43 @@ func computeWantedControllerRegistrationNames(
 		}
 	}
 
-	for _, requiredExtension := range wantedKindTypeCombinations.UnsortedList() {
-		names, ok := kindTypeToControllerRegistrationNames[requiredExtension]
+	installedAndRequired := installedAndRequiredRegistrationNames(controllerInstallationList, seedObjectMeta.Name)
+	if seedObjectMeta.DeletionTimestamp == nil {
+		wantedControllerRegistrationNames.Insert(installedAndRequired.List()...)
+	}
+
+	for _, wantedExtension := range wantedKindTypeCombinations.UnsortedList() {
+		names, ok := kindTypeToControllerRegistrationNames[wantedExtension]
 		if !ok {
-			return nil, fmt.Errorf("need to install an extension controller for %q but no appropriate ControllerRegistration found", requiredExtension)
+			return nil, fmt.Errorf("need to install an extension controller for %q but no appropriate ControllerRegistration found", wantedExtension)
+		}
+		if seedObjectMeta.DeletionTimestamp != nil {
+			for _, name := range names {
+				if installedAndRequired.Has(name) {
+					wantedControllerRegistrationNames.Insert(name)
+				}
+			}
+		} else {
+			wantedControllerRegistrationNames.Insert(names...)
```

Comment:
Suggestion to reduce the nesting and improve readability:

```suggestion

		// <- Maybe you can add a comment here and explain what is going on and why we need this special behaviour
		if seedObjectMeta.DeletionTimestamp == nil {
			wantedControllerRegistrationNames.Insert(names...)
			continue
		}
		
		for _, name := range names {
			if installedAndRequired.Has(name) {
				wantedControllerRegistrationNames.Insert(name)
			}
		}
```